.github/workflows/flake-check.yml

```
@@ -0,0 +1,24 @@
+name: Flake Check
+
+on:
+  push:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths-ignore:
+      - "flake.lock"
+
+jobs:
+  flake-check:
+    runs-on: ubuntu-latest
```

lets add a matrix to split up the runner so we don't run out of space
```suggestion
    runs-on: ubuntu-latest
        strategy:
        matrix:
          rust:
            - msrv
            - stable
            - nightly
```